### PR TITLE
Streamlabs integration

### PR DESCRIPTION
**Pull request by Magentah**
This is a working base for streamlabs integration so that work can begin on reacting to streamlabs events.

### CODE REVIEW

File: server/src/services/streamlabsService.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        try {
+            Logger.info(LogType.Streamlabs, `Connecting to streamlabs socket with token: ${socketToken}`);
+            this.websocket = io(`${Constants.StreamlabsSocketEndpoint}?token=${socketToken}`, {
+                transports: ["websocket"],
+                reconnection: true,
+            });
+        } catch (err) {
+            Logger.err(LogType.Streamlabs, "Error", err);
```
Comment (floele) [inline server/src/services/streamlabsService.ts:74]: If you just catch and continue, doesn't the code below fail?

File: server/src/models/user.ts [hunk 2]
```diff
@@ -6,7 +6,6 @@ export default interface IUser {
     id?: number;
     username: string;
     idToken?: string;
-    accessToken?: string;
```
[the remark this answers is not in the document]
Comment (Magentah) [reply server/src/models/user.ts:9]: I'm not sure why we need to save it, it wasn't being used anywhere that I could see.

File: server/src/services/streamlabsService.ts (hunk 1 above)
Comment (Magentah) [reply server/src/services/streamlabsService.ts:74]: The catch is for connection errors, the websocket will still be a valid object afterwards.

File: server/src/models/user.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (floele) [reply server/src/models/user.ts:9]: I think we need it in `fetchUserProfile()`. It's getting a principal from `getTwitchUserPrincipal(user)`, which gets access and refresh token from the users table. It is then put into the authorisation header in `buildHeaderFromUserPrincipal()`, see `"Authorization": Bearer ${ctx.accessToken},`.
So this should no longer be true with the merge of #18

[the remark this answers is not in the document]
Comment (Magentah) [reply server/src/models/user.ts]: Added this back in so that it can be updated in a separate PR.

**Review by floele — APPROVED** [review]

**Pull request merged**